appinventor/appengine/src/com/google/appinventor/client/editor/simple/components/MockCircle.java

```diff
@@ -219,6 +219,8 @@ protected native void addToMap(JavaScriptObject map)/*-{
     this.@com.google.appinventor.client.editor.simple.components.MockMapFeatureBase::setNativeTooltip(*)(
       this.@com.google.appinventor.client.editor.simple.components.MockMapFeatureBase::getTooltip()()
     );
+    var isVisible = this.@com.google.appinventor.client.editor.simple.components.MockMapFeatureBase::getVisibleProperty()();
+    if(!isVisible) map.removeLayer(circle);
```

Comment:
Code style: Space between `if` and the opening parens. You should also have the body of the if statement on its own line and within curly braces. Make similar updates to the other places you've added this check in this PR.

Reply:
Done!